onnxruntime/core/providers/dnnl/dnnl_custom_threadpool.h

```diff
@@ -0,0 +1,121 @@
+#pragma once
+#include "dnnl_threadpool_iface.hpp"
+#include "core/platform/threadpool.h"
+#include "core/platform/EigenNonBlockingThreadPool.h"
+
+// oneDNN limitations using eigen
+#if !(EIGEN_WORLD_VERSION == 3 && EIGEN_MAJOR_VERSION == 3) && defined(DNNL_EIGEN_THREAD)
+#error Unsupported Eigen version (need 3.3.x)
+#endif
+
+// To improve readability
+using OneDnnlThreadPoolIface = dnnl::threadpool_interop::threadpool_iface;
```

Comment:
Can you use dnnl instead of onednnl in the names? Our code still uses dnnl. At some point, we'll rename all to onednn.